data/Store.js

```diff
@@ -947,6 +965,25 @@ function forIn(obj, fn) {
     }
 }
 
+/**
+ * Modes for how record field values should be considered when filtering.
+ * @readonly
+ * @enum {string}
+ */
+export const FilterValueMode = Object.freeze({
+    /** The committed value is used to test the filter. */
+    COMMITTED: 'committed',
+
+    /** The current value is used to test the filter. */
+    CURRENT: 'current',
+
+    /**
+     * Both the committed and current values are used to test the filter and the record passes if
+     * either value passes.
+     */
+    ANY: 'any'
+});
```

Comment:
Could we use the comments in this enum to offer a little bit of context or advice for when / why you might opt for one mode vs. another?  Mainly to point at inline editing as the case where you commonly want the default `COMMITTED` behavior. 

If there is a clear / typical use case for the `CURRENT` we could call that out as well.

Reply:
Actually after playing around with it I think `ANY` is the best mode for inline editing, as it handles the 2 common inline editing use cases with filtering:

1. User modifies quantity field on record to equal 100
2. User adds field filter quantity==100 - modified record is still visible since current quantity == 100, even though committed quantity != 100
3. User commits record - record still visible

1. User adds field filter quantity==100
2. User modifies quantity field on record to equal 50 - record still visible since committed quantity == 100
3. User commits modified record - record disappears since committed and current quantity == 100

I think a case could be made for `ANY` to be the default mode actually, have a hard time coming up with uses for the other modes but felt it was best to leave it flexible for apps to configure so not really sure what we could recommend those modes for in the comments ...